packages/run-protocol/src/vaultFactory/types.js

```diff
@@ -34,8 +37,8 @@
 
 /**
  * @typedef  {Object} VaultFactoryPublicFacet - the public facet
- * @property {() => Promise<Invitation>} makeLoanInvitation
- * @property {() => Promise<Invitation>} makeVaultInvitation
+ * @property {() => Promise<Invitation<VaultKit>>} makeLoanInvitation
+ * @property {() => Promise<Invitation<VaultKit>>} makeVaultInvitation
```

Comment:
Should one of these be deprecated?   Ah, I see it in `vaultFactory.js`. Shouldn't it also be annotated in the type declaration?

Reply:
Ah, I found a solution with using the implementation to define the type. The TS and the IDE can see what property is deprecated:
<img width="514" alt="Screen Shot 2022-02-28 at 7 41 27 AM" src="https://user-images.githubusercontent.com/21505/156012621-1a621e98-5b29-43b4-ba3e-d583cbfbac92.png">

I also think it's a better pattern to define types as they are returned by maker functions (like class constructors).

Reply:
> I also think it's a code quality and velocity improvement to define types as they are returned by maker functions (like class constructors).

Sorry, I didn't get this.  Which part is an improvement? How does it help? Are we using something like class constructors?

Reply:
I'll try to give more context:

`makeFoo` is like a `Foo` class constructor. If we weren't programming in SES we'd be using classes. When you have types for classes, you rely on the class definition as the type.

We have been using a typedef to say what `Foo` is and then saying that `makeFoo` returns that. That's kind of like defining an interface that only ever gets implemented by one class. Instead, we can use `ReturnType` to say "type `Foo` is whatever `makeFoo` returns".

That's an improvement because:
- when you see `Foo` you can command-clicked once or twice to get to the actual implementation
- changes to what `makeFoo` returns don't require also revising the interface; what `makeFoo` returns is authoritative for what type `Foo` is